catroid/src/main/java/org/catrobat/catroid/content/bricks/SpeakBrickTools.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.content.bricks;
+
+import android.content.Context;
+import android.database.DataSetObserver;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Spinner;
+import android.widget.SpinnerAdapter;
+import android.widget.TextView;
+
+import org.catrobat.catroid.CatroidApplication;
+import org.catrobat.catroid.R;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import static org.catrobat.catroid.common.Constants.AVAILABLE_LOCALES_TTS;
+
+public class SpeakBrickTools {
+	protected static Locale locale;
+
+	public void setLocale(Locale mlocale) {
+		locale = mlocale;
+	}
+
+	public static Locale getLocale() {
+		return locale;
+	}
+
+	static void setSpeakSpinner(Context context, View view, boolean isPrototypeView) {
```

Comment:
please split in two methods, eg. setupSpinner and setItemSelectedListener and remove the isPrototypeView parameter. either call one or both methods depending on what you need.